src/satosa/micro_services/processors/scope_extractor_processor.py

```diff
@@ -0,0 +1,40 @@
+from ..attribute_processor import AttributeProcessorError, AttributeProcessorWarning
+from .base_processor import BaseProcessor
+
+
+CONFIG_KEY_MAPPEDATTRIBUTE = 'mapped_attribute'
+CONFIG_DEFAULT_MAPPEDATTRIBUTE = ''
+
+
+class ScopeExtractorProcessor(BaseProcessor):
+    """
+    Extracts the scope from a scoped attribute and maps that to
+    another attribute
+
+    Example configuration:
+    module: satosa.micro_services.attribute_processor.AttributeProcessor
+    name: AttributeProcessor
+    config:
+      process:
+      - attribute: scoped_affiliation
+        processors:
+        - name: ScopeExtractorProcessor
+          module: satosa.micro_services.processors.scope_extractor_processor
+          mapped_attribute: domain
+    """
+    def process(self, internal_data, attribute, **kwargs):
+        mapped_attribute = kwargs.get(CONFIG_KEY_MAPPEDATTRIBUTE, CONFIG_DEFAULT_MAPPEDATTRIBUTE)
+        if mapped_attribute is None or mapped_attribute == '':
+            raise AttributeProcessorError("The mapped_attribute needs to be set")
+
+        attributes = internal_data.attributes
+        values = attributes.get(attribute, [None])
+        if not values:
+            raise AttributeProcessorWarning("Cannot apply scope_extractor to {}, it has no values".format(attribute))
+        if not any('@' in val for val in values):
+            raise AttributeProcessorWarning("Cannot apply scope_extractor to {}, it's values are not scoped".format(attribute))
```

Comment:
This will not stop execution so 
```python
any('@' in val for val in values)
```
 will raise a TypeError. 
It would make more sense to return an empty list as a default value of  `get()` in 

```python
attributes.get(attribute, [None])
```

Reply:
Sure, that would do 👍